bin/run-command.sh

```diff
@@ -0,0 +1,84 @@
+#     e.g. To run the command `db-migrate-up` with no arguments, set
+#     COMMAND='["db-migrate-up"]'
+#     e.g. To run the command `echo "Hello, world"` set
+#     COMMAND='["echo", "Hello, world"]')
+# -----------------------------------------------------------------------------
+set -euo pipefail
+
+APP_NAME="$1"
+ENVIRONMENT="$2"
+COMMAND="$3"
+ENVIRONMENT_VARIABLES=${4:-""}
+
+echo "==============="
+echo "Running command"
+echo "==============="
+echo "Input parameters"
+echo "  APP_NAME=$APP_NAME"
+echo "  ENVIRONMENT=$ENVIRONMENT"
+echo "  COMMAND=$COMMAND"
+echo "  ENVIRONMENT_VARIABLES=$ENVIRONMENT_VARIABLES"
+echo
+
+# Use the same cluster, task definition, and network configuration that the application service uses
+CLUSTER_NAME=$(terraform -chdir=infra/$APP_NAME/service output -raw service_cluster_name)
+SERVICE_NAME=$(terraform -chdir=infra/$APP_NAME/service output -raw service_name)
+
+SERVICE_TASK_DEFINITION_ARN=$(aws ecs describe-services --no-cli-pager --cluster $CLUSTER_NAME --services $SERVICE_NAME --query "services[0].taskDefinition" --output text)
+# For subsequent commands, use the task definition family rather than the service's task definition ARN
+# because in the case of migrations, we'll deploy a new task definition revision before updating the
+# service, so the service will be using an old revision, but we want to use the latest revision.
+TASK_DEFINITION_FAMILY=$(aws ecs describe-task-definition --no-cli-pager --task-definition $SERVICE_TASK_DEFINITION_ARN --query "taskDefinition.family" --output text)
+
+NETWORK_CONFIG=$(aws ecs describe-services --no-cli-pager --cluster $CLUSTER_NAME --services $SERVICE_NAME --query "services[0].networkConfiguration")
+CURRENT_REGION=$(./bin/current-region.sh)
+AWS_USER_ID=$(aws sts get-caller-identity --no-cli-pager --query UserId --output text)
+
+ENVIRONMENT_OVERRIDES=""
+if [ ! -z "$ENVIRONMENT_VARIABLES" ]; then
+  ENVIRONMENT_OVERRIDES="\"environment\": $ENVIRONMENT_VARIABLES,"
+fi
+CONTAINER_NAME=$(aws ecs describe-task-definition --task-definition $TASK_DEFINITION_FAMILY --query "taskDefinition.containerDefinitions[0].name" --output text)
+OVERRIDES=$(cat << EOF
+{
+  "containerOverrides": [
+    {
+      $ENVIRONMENT_OVERRIDES
+      "name": "$CONTAINER_NAME",
+      "command": $COMMAND
+    }
+  ]
+}
+EOF
+)
+
+AWS_ARGS=(
+  ecs run-task
+  --region=$CURRENT_REGION
+  --cluster=$CLUSTER_NAME
+  --task-definition=$TASK_DEFINITION_FAMILY
+  --started-by=$AWS_USER_ID
+  --launch-type=FARGATE
+  --platform-version=1.4.0
+  --network-configuration "$NETWORK_CONFIG"
+  --overrides "$OVERRIDES"
+)
+echo "Running AWS CLI command"
+printf " ... %s\n" "${AWS_ARGS[@]}"
+echo
+TASK_ARN=$(aws --no-cli-pager "${AWS_ARGS[@]}" --query "tasks[0].taskArn" --output text)
+echo
+echo "Waiting for task to stop"
+echo "  TASK_ARN=$TASK_ARN"
+echo
+aws ecs wait tasks-stopped --region $CURRENT_REGION --cluster $CLUSTER_NAME --tasks $TASK_ARN
```

Comment:
Small tweak that we have that I think might be helpful. It shows the status from the command, and if the exit code is missing (meaning it is still running) or not 0, it throws an error so that the workflow knows there was an issue. The wait command has a limit of 10 minutes, so we have it wait twice to try and catch any long migrations, but that could be overkill, and I think is fine not being part of the template

```suggestion
aws ecs wait tasks-stopped --region $CURRENT_REGION --cluster $CLUSTER_NAME --tasks $TASK_ARN

TASK_STATUS=$(aws ecs describe-tasks --cluster $CLUSTER_NAME --task $TASK_ARN | jq -r '.tasks[]')
LAST_STATUS=$(echo $TASK_STATUS | jq '.containers[].lastStatus' | tr -d '\"')
echo "Task status: $LAST_STATUS"

EXIT_CODE=$(echo $TASK_STATUS | jq '.containers[].exitCode')

# If there is no exit code, or the exit code is not 0
if [[ "$EXIT_CODE" == "null" || "$EXIT_CODE" != 0 ]]; then
  STOPPED_REASON=$(echo "$TASK_STATUS" | jq '.stoppedReason')
  echo "ECS task failed:" >&2
  echo "$STOPPED_REASON" >&2
  exit 1
fi
```

Reply:
Ok I made the changes! PTAL!

I also realized I forgot to push the example app changes, so just did that.